esp/services/ws_workunits/ws_workunitsQuerySets.cpp

```diff
@@ -766,11 +766,21 @@ class QueryFileCopier
         if (queryname && *queryname)
             queryname = queryid.append(queryname).append(".0").str(); //prepublish dummy version number to support fuzzy match like queries="myquery.*" in package
         files->addFilesFromQuery(cw, pm, queryname);
+        Owned<IDFUhelper> helper = createIDFUhelper();
+#ifdef _CONTAINERIZED
+        //Find the first data plane
+        Owned<IPropertyTreeIterator> dataPlanes = queryGlobalConfig().getElements("storage/planes[labels='data']");
+        if (!dataPlanes->first())
+            throwUnexpectedX("No default data plane defined");
```

Comment:
What happens if the file is already local on the target plane?  Should this not be called?  Will this all work fine if it is already there - I need to test.

Reply:
tested and seems to work